### ISSUE TITLE
[REVIEW]: BESOS: Building and Energy Systems Optimization and Surrogate-modelling

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#2677]
**Submitting author:** @TheoChristiaanse (<a href="http://orcid.org/0000-0003-4082-3338">Theodor Victor Christiaanse</a>)
**Repository:** <a href="https://gitlab.com/energyincities/besos/" target ="_blank">https://gitlab.com/energyincities/besos/</a>
**Version:** 2.1.3
**Editor:** @sjpfenninger
**Reviewer:** @fneum, @willu47
**Archive:** <a href="https://doi.org/10.5281/zenodo.4667614" target="_blank">10.5281/zenodo.4667614</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/e8eb9f8e7f75452569aaf37c7005e640/status.svg)](https://joss.theoj.org/papers/e8eb9f8e7f75452569aaf37c7005e640)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/e8eb9f8e7f75452569aaf37c7005e640"><img src="https://joss.theoj.org/papers/e8eb9f8e7f75452569aaf37c7005e640/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/e8eb9f8e7f75452569aaf37c7005e640/status.svg)](https://joss.theoj.org/papers/e8eb9f8e7f75452569aaf37c7005e640)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://gitlab.com/energyincities/besos/" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@fneum & @willu47, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @sjpfenninger know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://gitlab.com/energyincities/besos/">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@TheoChristiaanse) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://gitlab.com/energyincities/besos/">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@TheoChristiaanse) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @fneum, @willu47 it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02677/joss.02677/10.21105.joss.02677.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02677/joss.02677/10.21105.joss.02677.pdf) :page_facing_up: :point_left:

**Comment by fneum** [comment]
## Housekeeping

This is my first review for JOSS, so any feedback is appreciated.

I have posted bigger thematic blocks as issues at https://gitlab.com/energyincities/besos/-/issues and referenced those also below.

Congrats on your work already! The review may seem critical regarding some aspects, but I hope it's always constructive! There are also many suggestions which are not strictly essential.

## General

This is a very useful library focussing on user-friendly interfacing and data structures, rather than new functionality.
It provides interfaces to common building energy optimisation tools.
The interfaces regard both ranges of input parameters and convenient collection of output data.
Moreover, it includes functions for sampling from the parameter space.
This facilitates building surrogate models with machine learning techniques.
Such surrogate models have similar wrapping as the interfaces to full-scale building models.
Thereby, they can conveniently be used as substitutes for computationally fast evaluations.
This is great because it allows thoroughly exploring the design space.
Overall, also the execution of parametric runs of tools like EnergyPlus and EnergyHub is advanced.

## License

- checked https://gitlab.com/energyincities/besos/-/blob/master/LICENSE
- GNU GPL v3

## Repository Link

- checked https://gitlab.com/energyincities/besos/

## Substantial Scholarly Effort

Without a doubt:

- The `git` history dates back >2 years. 
- Includes substantial amounts of maintained code.
- Multiple authors involved.

## Contribution and Authorship

All listed authors have contributed to the code.

One question I had was whether Paul Westermann is listed as first author for a particular reason while having contributed only few commits in the history? Maybe he is the initial developer that doesn't show in the git history.

There were some further contributors in the past year not listed as authors:

Evan Guo [NA, 78 commits]
Chase [NA, 50 commits]
Goh Sato [NA, 46 commits]
jjcomish@shaw.ca [NA, 28 commits]
C Kwan [NA, 26 commits]

It's up to you who you want to take onboard, but I just wanted to let you consider it.

## Installation

https://gitlab.com/energyincities/besos/-/issues/46

## General Remarks on Code / Repository

https://gitlab.com/energyincities/besos/-/issues/47

## Documentation

https://gitlab.com/energyincities/besos/-/issues/48

## Paper

https://gitlab.com/energyincities/besos/-/issues/49

## Examples

general: https://gitlab.com/energyincities/besos/-/issues/50

individual: https://gitlab.com/energyincities/besos/-/issues/51

**Comment by sjpfenninger** [comment]
@fneum Thanks for this careful and extensive review!

**Comment by sjpfenninger** [comment]
@willu47 Are you able to complete your review in the near future?

**Comment by willu47** [comment]
Hi @sjpfenninger - I've submitted a bunch of issues over on the besos Gitlab - will summarise here shortly.

**Comment by willu47** [comment]
The review of @fneum is very comprehensive, and I have tried not to overlap with the recommendations from that too much.
I've posted a number of issues over on the Gitlab repository used by the team to manage the source code.

In short, this package makes a useful contribution, bringing together a number of existing tools to extend and enhance building energy modelling software with methods to explore uncertainty, optimisation and interoperability with e.g. a district heating simulation package.

My main comments as follows with recommendations to be addressed:

- tidy up the installation instructions and process [[issue]](https://gitlab.com/energyincities/besos/-/issues/57) and fix errors with dependency upon particular version of EnergyPlus [[issue]](https://gitlab.com/energyincities/besos/-/issues/54)
- provide guidance to the wider community on how to contribute to the package. At present there's some developer docs in the readme. I'd recommend a `contribution.md` file in the root of the repo containing the detailed instructions, and for this to be included in the read-the-docs via an include and referenced from the readme.  It's also worth considering adopting a code-of-conduct. [[issue]](https://gitlab.com/energyincities/besos/-/issues/56)
- The focus of the paper is not clear [[issue]](https://gitlab.com/energyincities/besos/-/issues/58). Personally, I think that the package alone deserves to be the sole focus of the JOSS paper, and that the "platform", which seems to be just a JupyterHub that hosts the notebooks, could be a separate enterprise, in a separate repo, and another paper.  It's worth mentioning the platform for context, but at the moment it seems to distract rather than support the rationale for the paper.
- Recommend reorganising the structure of the Python package [[issue]](https://gitlab.com/energyincities/besos/-/issues/55)
- Otherwise, I support the recommendations of @fneum, particularly documentation (which I haven't touched on substantially).

Happy to answer any questions and support the authors in their process of addressing these comments.

**Comment by sjpfenninger** [comment]
Thanks @willu47 for your careful review!

**Comment by TheoChristiaanse** [comment]
Thank you for the review! We have been working on improving the repo. Will report back when we have delt with the issues.

**Comment by sjpfenninger** [comment]
@TheoChristiaanse I hope work is proceeding well. Let me know if anything is unclear while working through the reviewer comments or whether you have queries about how to address them.

**Comment by sjpfenninger** [comment]
@TheoChristiaanse Can you give an update on how things are going?

**Comment by TheoChristiaanse** [comment]
> 
> 
> @TheoChristiaanse Can you give an update on how things are going?

Hello @sjpfenninger 



We are done changing most of the different issues. The eventual branch we will use for the new version is here:

https://gitlab.com/energyincities/besos/-/tree/pre2.0

The new readthedocs can be found here:

https://besos.readthedocs.io/en/jossfixes/

The website has also changed.

https://besos.uvic.ca/

Things that we didn't get to are captured in new issues posted on the issues board. 

let me know we have completed the review.

Thank you again for the all the hard work on the review! We have improved our repository a lot!

**Comment by TheoChristiaanse** [comment]
We have started also moving the examples to a different repo. This should greatly reduce the size of the repo lowering the core to around 500kb. I will update the docs as we move forward.

**Comment by TheoChristiaanse** [comment]
@sjpfenninger Happy new year! I hope you are doing well.

Do you know how to progress with the paper/repo review? The latest version we have launched 1.7.1 contains most of the corrections for the JOSS paper. The latest version pre2.0 will be launched soon with all the corrections applied.

**Comment by sjpfenninger** [comment]
@TheoChristiaanse If I understand correctly you are still working on addressing reviewer comments which will be done in version pre2.0 -- can you update here on the thread once that is ready? Then we can proceed.

**Comment by sjpfenninger** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02677/joss.02677/10.21105.joss.02677.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02677/joss.02677/10.21105.joss.02677.pdf) :page_facing_up: :point_left:

**Comment by sjpfenninger** [comment]
@fneum @willu47 Could you have a look to see whether you are happy with how your comments have been addressed?

@TheoChristiaanse Great, thanks for the revisions!

The first paragraph in the paper has a dangling incomplete sentence, can you fix this?

Right off the bat I still see broken links. For example at the very bottom of the README you point to https://gitlab.com/energyincities/besos/-/blob/master/examples/ExamplesOverview.ipynb which doesn't exist. In the documentation's index page you point to https://besos.readthedocs.io/en/stable/ExamplesOverview.html and https://besos.readthedocs.io/en/stable/installation which don't exist. This is not an exhaustive list, just ones I immediately spotted. Can you carefully review your README, documentation, and example files, to fix such broken links?

**Comment by fneum** [comment]
Well done! Both repository and documentation look much tidier now! Also the EnergyPlus installation in a custom directory worked without problems for me now. Excellent! Almost all aspects from my initial review were adequately addressed. In addition to @sjpfenninger's remarks, only the following aspects still need your attention before we can wrap-up:

### Re-integration of examples directory (links/submodule)

The broken links mentioned are due to the missing/moved examples directory and seems to apply to all links from readthedocs to the examples page. In https://gitlab.com/energyincities/besos/-/issues/50 you said you were going to include this directory as a submodule but you may have missed doing that. Can you fix that so that the examples are easily accessible from the documentation? On a side note: even though you will include it as submodule to the main repository, https://gitlab.com/energyincities/besos-examples should get a license.

### API modules in documentation

Is it intended that most of the API modules autodocs are empty in the documentation (e.g. https://besos.readthedocs.io/en/stable/modules/evaluator.html)?

### Nit-picking

Setting upper bounds in requirements is better than pinning versions if newer versions are the only issue: I'd recommend: `xlrd<=1.2.0` and `matplotlib<=3.2.2`.

The reference reference https://doi.org/10.11826868/25222708.2019.211232 seems to miss the date which should be 2019?

**Comment by TheoChristiaanse** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02677/joss.02677/10.21105.joss.02677.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02677/joss.02677/10.21105.joss.02677.pdf) :page_facing_up: :point_left:

**Comment by sjpfenninger** [comment]
@TheoChristiaanse Can you fix the broken links and address the two major points from @fneum (Re-integration of examples directory (links/submodule), API modules in documentation) -- you may want to consider his point about upper bounds rather than pinning but I leave it up to you. And can you fix the missing date in the reference 10.11826868/25222708.2019.211232.

**Comment by sjpfenninger** [comment]
@willu47 The issues you opened are all closed, can you let us know here whether you are happy with how your comments have been addressed?

**Comment by willu47** [comment]
@sjpfenninger @TheoChristiaanse apologies for the delay in responding. I am happy with the responses to the issues and recommend acceptance for publication.

**Comment by sjpfenninger** [comment]
@TheoChristiaanse Can you let me know when you have addressed [my points from above](https://github.com/openjournals/joss-reviews/issues/2677#issuecomment-774022267)?

**Comment by TheoChristiaanse** [comment]
> @TheoChristiaanse Can you fix the broken links and address the two major points from @fneum (Re-integration of examples directory (links/submodule), API modules in documentation) -- you may want to consider his point about upper bounds rather than pinning but I leave it up to you. And can you fix the missing date in the reference 10.11826868/25222708.2019.211232.

- We have removed the pinning on most requirements
- The solution we found to import the examples into the docs is to download the jupyter notebooks. Then convert them to rst while replacing links to the internal docs is those jupyter notebooks have been imported. Not all jupyter notebooks are imported because some are specific to the BESOS platform.
- Fixed the year date.

**Comment by fneum** [comment]
LGTM

**Comment by danielskatz** [comment]
@sjpfenninger - a gentle ping...

**Comment by sjpfenninger** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02677/joss.02677/10.21105.joss.02677.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02677/joss.02677/10.21105.joss.02677.pdf) :page_facing_up: :point_left:

**Comment by sjpfenninger** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1080/09613210701325883 is OK
- 10.1016/j.apenergy.2020.115563 is OK
- 10.26868/25222708.2019.211232 is OK
- 10.1016/j.apenergy.2020.114715 is OK
- 10.1145/3360322.3360995 is OK

MISSING DOIs

- 10.1016/j.energy.2014.06.029 may be a valid DOI for title: New formulations of the ‘energy hub’model to address operational constraints
- 10.1016/j.enbuild.2019.01.048 may be a valid DOI for title: Building energy optimization: An extensive benchmark of global search algorithms

INVALID DOIs

- None
```

**Comment by sjpfenninger** [comment]
@TheoChristiaanse: Almost there!

The very first link on https://besos.readthedocs.io/en/stable/ is still broken: [examples overview page](https://besos.readthedocs.io/en/stable/ExamplesOverview.html). Further down there are more broken links, e.g. https://besos.readthedocs.io/en/stable/installation.

Could you please double-check the links in the docs, especially on the index page where people are most likely to start looking?

Can you also add a DOI to all references in the bibliography where a DOI exists? (at least two are missing, see above comment from whedon)

**Comment by sjpfenninger** [comment]
@TheoChristiaanse could you also point me to the answer to @fneum's questions about authorship of the paper? I cannot see it here nor in the review issues fneum created in your repo.

**Comment by sjpfenninger** [comment]
@TheoChristiaanse You have multiple published papers without a DOI, for example `evins2014new`. Can you fix those?

**Comment by TheoChristiaanse** [comment]
I've added more doi's.

on the authors:

@fneum The main author of the paper "Paul Westernmann" is someone who wrote the first draft of the paper as well as contributed to the conceptualization of the main objects( evaluator, parameter, sampling), adaptive sampling, and code review. We continuously have new students under the supervision of me, Dr Evins or Mr. Kovacs join help work on the repo. The list of the authors is a core group of students, researchers, staff and faculty that will determine the direction of the repository, or have made a significant contribution to the repository.

**Comment by sjpfenninger** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02677/joss.02677/10.21105.joss.02677.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02677/joss.02677/10.21105.joss.02677.pdf) :page_facing_up: :point_left:

**Comment by sjpfenninger** [comment]
@TheoChristiaanse thanks for the DOIs and the authorship clarification. I've suggested [some minor language/clarity edits](https://gitlab.com/energyincities/besos/-/merge_requests/125). Please consider those and merge as appropriate. Then we're ready to accept: make a tagged release on GitHub and archive that release, for example on Zenodo or figshare, and report the version number and archive DOI back here in the review issue.

**Comment by TheoChristiaanse** [comment]
> 
> 
> @TheoChristiaanse thanks for the DOIs and the authorship clarification. I've suggested [some minor language/clarity edits](https://gitlab.com/energyincities/besos/-/merge_requests/125). Please consider those and merge as appropriate. Then we're ready to accept: make a tagged release on GitHub and archive that release, for example on Zenodo or figshare, and report the version number and archive DOI back here in the review issue.

Hi @sjpfenninger We have released 2.1 which has all the JOSS updates and should be a final version that can be accepted. Since we developed on Gitlab. Is there a particular site that we should use that works well with Gitlab? Or should we fork to Github to archive here? then connect with Zenodo or figshare?

**Comment by TheoChristiaanse** [comment]
@sjpfenninger Hi, I just looked at the Zenodo service, we can also upload the files from the GitLab manually. We can keep it up to date by uploading when there are new releases. However, it would not have this link with Github.  Would this also be sufficient?

**Comment by sjpfenninger** [comment]
You can manually upload a copy of the repository to Zenodo. You do not need to update it after the paper is accepted. What is important is that the title and authors in the Zenodo deposition match those of the paper.

**Comment by TheoChristiaanse** [comment]
Hi @sjpfenninger 

I've uploaded the files to https://zenodo.org/record/4631191

**Comment by sjpfenninger** [comment]
Thank you. Can you look at [my suggested edits](https://gitlab.com/energyincities/besos/-/merge_requests/125) also? I see those remain untouched. There are some minor grammar/missing word fixes.

**Comment by TheoChristiaanse** [comment]
Hi @sjpfenninger 

we have merged you corrections to the paper. 

I also, made it into a new version 2.1.1, and you can find that it also archived here: https://zenodo.org/record/4633738

Best,
Theo

**Comment by sjpfenninger** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02677/joss.02677/10.21105.joss.02677.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02677/joss.02677/10.21105.joss.02677.pdf) :page_facing_up: :point_left:

**Comment by sjpfenninger** [comment]
@whedon set v2.1.1 as version

**Comment by whedon** [comment]
OK. v2.1.1 is the version.

**Comment by sjpfenninger** [comment]
@whedon set 10.5281/zenodo.4633738 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4633738" target="_blank">10.5281/zenodo.4633738</a> is the archive.

**Comment by sjpfenninger** [comment]
@TheoChristiaanse Great thanks, we're ready here! Just one minor point about the Zenodo archive, it says "License: Other (non-commercial)", but I think you ought to be able to change that to GNU GPLv3 which is what the LICENSE file says.

**Comment by sjpfenninger** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1080/09613210701325883 is OK
- 10.1016/S0378-7788(00)00114-6 is OK
- 10.1016/j.energy.2014.06.029 is OK
- 10.1016/j.enbuild.2019.05.057 is OK
- 10.1016/j.enbuild.2019.01.048 is OK
- 10.1016/j.apenergy.2020.115563 is OK
- 10.26868/25222708.2019.211232 is OK
- 10.1016/j.apenergy.2020.114715 is OK
- 10.1145/3360322.3360995 is OK

MISSING DOIs

- 10.1016/j.seta.2021.101108 may be a valid DOI for title: Optimal storage systems for residential energy1systems in British Columbia

INVALID DOIs

- 10.5555/1953048.2078195 is INVALID
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2160

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2160, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by sjpfenninger** [comment]
Not sure what's going on with DOI 10.5555/1953048.2078195, https://dl.acm.org/doi/10.5555/1953048.2078195 suggests it exists, but it doesn't resolve...

**Comment by sjpfenninger** [comment]
Anyway, thanks @willu47 and @fneum for your extensive reviews and @TheoChristiaanse for working on incorporating the changes!

**Comment by TheoChristiaanse** [comment]
> Not sure what's going on with DOI 10.5555/1953048.2078195, https://dl.acm.org/doi/10.5555/1953048.2078195 suggests it exists, but it doesn't resolve...

It works on my end.

**Comment by TheoChristiaanse** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
I'm sorry @TheoChristiaanse, I'm afraid I can't do that. That's something only editor-in-chiefs are allowed to do.

**Comment by TheoChristiaanse** [comment]
@whedon accept

**Comment by whedon** [comment]
I'm sorry @TheoChristiaanse, I'm afraid I can't do that. That's something only editors are allowed to do.

**Comment by TheoChristiaanse** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02677/joss.02677/10.21105.joss.02677.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02677/joss.02677/10.21105.joss.02677.pdf) :page_facing_up: :point_left:

**Comment by TheoChristiaanse** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02677/joss.02677/10.21105.joss.02677.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02677/joss.02677/10.21105.joss.02677.pdf) :page_facing_up: :point_left:

**Comment by TheoChristiaanse** [comment]
@whedon generate pdf

**Comment by TheoChristiaanse** [comment]
I've removed the doi from scikit learn. I used the bibtex from the scikit learn website: https://scikit-learn.org/stable/about.html#citing-scikit-learn

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02677/joss.02677/10.21105.joss.02677.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02677/joss.02677/10.21105.joss.02677.pdf) :page_facing_up: :point_left:

**Comment by TheoChristiaanse** [comment]
https://zenodo.org/record/4638214 


I've uploaded a new version 2.1.2 with the changes. https://gitlab.com/energyincities/besos/-/tags/v2.1.2

Azin's paper has also been published now, so I updated the bibtex again to include the new information.

**Comment by kthyng** [comment]
@willu47 and @sjpfenninger I see there is one unchecked box above. Is this meaningful? All the boxes need to be checked to finish this process.

**Comment by willu47** [comment]
@kthyng - final box ticked!

**Comment by kthyng** [comment]
@willu47 thank you!

**Comment by kthyng** [comment]
Zenodo archive looks good!

**Comment by kthyng** [comment]
@TheoChristiaanse Your paper looks good but the references should be cleaned up a bit. Please look through for issues yourself, but a few I see:
* Preserve capitalization in the .bib file by placing {} around the appropriate words. For example, British Columbia should be capitalized in the first reference
* need a space in Evins et al.

You don't need to make a new version for edits to the paper unless you want to. I'll tag a new version at v2.1.2 unless you want to release a new one in which case I'll use that.

**Comment by danielskatz** [comment]
@TheoChristiaanse - have you made the changes @kthyng requested?

**Comment by TheoChristiaanse** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.02677/joss.02677/10.21105.joss.02677.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.02677/joss.02677/10.21105.joss.02677.pdf) :page_facing_up: :point_left:

**Comment by TheoChristiaanse** [comment]
Hello @danielskatz @kthyng 

I went through the bibtex again, I found one more issue, but the rest look good to me. I've updated the version again to keep track of the changes.

https://gitlab.com/energyincities/besos/-/tags/2.1.3

and uploaded this version also to the zenodo.

https://zenodo.org/record/4638214

let me know if there are any other issues.

**Comment by danielskatz** [comment]
@kthyng - this looks good to me - are we ready to accept?

**Comment by danielskatz** [comment]
@whedon set 2.1.3 as version

**Comment by whedon** [comment]
OK. 2.1.3 is the version.

**Comment by danielskatz** [comment]
@whedon set 10.5281/zenodo.4667614 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.4667614" target="_blank">10.5281/zenodo.4667614</a> is the archive.

**Comment by danielskatz** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1080/09613210701325883 is OK
- 10.1016/S0378-7788(00)00114-6 is OK
- 10.1016/j.seta.2021.101108 is OK
- 10.1016/j.energy.2014.06.029 is OK
- 10.1016/j.enbuild.2019.05.057 is OK
- 10.1016/j.enbuild.2019.01.048 is OK
- 10.1016/j.apenergy.2020.115563 is OK
- 10.26868/25222708.2019.211232 is OK
- 10.1016/j.apenergy.2020.114715 is OK
- 10.1145/3360322.3360995 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2200

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2200, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by danielskatz** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1379535926938992640) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2201
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.02677](https://doi.org/10.21105/joss.02677)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by danielskatz** [comment]
Congratulations to @TheoChristiaanse (Theodor Victor Christiaanse) and co-authors!!

And thanks to @fneum and @willu47 for reviewing and @sjpfenninger for editing!

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.02677/status.svg)](https://doi.org/10.21105/joss.02677)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.02677">
  <img src="https://joss.theoj.org/papers/10.21105/joss.02677/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.02677/status.svg
   :target: https://doi.org/10.21105/joss.02677
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.02677/status.svg)](https://doi.org/10.21105/joss.02677)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
